cutplace/fields.py

```diff
@@ -349,8 +368,28 @@ def validated_value(self, value):
             message = "value is %r but must be a decimal number: %s" % (value, error)
             raise errors.FieldValueError(message)
 
+        try:
+            self.valid_range.validate("value", result)
+        except errors.RangeValueError as error:
+            raise errors.FieldValueError(str(error))
+
         return result
 
+    def validate_length(self, value):
```

Comment:
`validated_length` can be reused as defined in `AbstractFieldFormat`. What we need is a `validated()` that converts `value` to a `decimal.Decimal`.